steel/docs/steel-commitments.md

```diff
@@ -0,0 +1,138 @@
+# Steel Commitments
+
+Steel retrieves state for a view call at a specific block, therefore it commits the relevant block data to the journal for validation on-chain; this data is known as a Steel commitment. 
+
+Concretely, a Steel commitment consists of a block identifier, and the block hash. To validate this block commitment on-chain, the block hash in the commitment is compared with the block hash available on-chain. If there is a discrepancy, there is no guarantee that the proof accurately reflects the correct blockchain state at the specified block.
+
+## Steel's Trust Anchor: The Blockhash
+
+Steel uses [revm] to generate an EVM execution environment, `EvmEnv` within the guest. When you create the `EvmEnv`, you can specify a block (the default is the latest block).
+
+```rust
+// Create an EVM environment from that provider 
+let mut env = EthEvmEnv::builder()
+.provider(provider.clone())
+.block_number(20842508)
+.build()
+.await?;
+```
+
+This block is used for RPC queries (i.e. `getStorageAt`) during the `preflight` call. Once the preflighting is done, `into_input` is called:
+
+```rust
+let evm_input = env.into_input().await?
+```
+
+During this `into_input` step, the preflight data is packed into a form that can be read and validated in the guest.
+
+Crucially, during the step, Steel will use the RPC call [eth\_getProof] for all accounts accessed during the preflight and the return data is combined into a sparse Merkle trie.
+
+Within the guest, calling `into_env` checks that all Merkle tries are consistent, have the correct root and **computes the corresponding block hash.**
+
+*This blockhash is Steel's trust anchor; it needs to be recomputed within the guest to validate the integrity of the RPC data.*
+
+```rust
+// Read the input from the guest environment.
+let input: EthEvmInput = env::read();
+let contract: Address = env::read();
+let account: Address = env::read();
+
+// Converts the input into a `EvmEnv`
+let env = input.into_env().with_chain_spec(&ETH_SEPOLIA_CHAIN_SPEC);
+```
+
+For the Steel commitment, it is this block hash, computed within the guest, that is committed to the journal:
+
+```rust
+let journal = Journal {
+	commitment: env.into_commitment(),
+	tokenAddress: contract,
+};
+env::commit_slice(&journal.abi_encode());
+```
+
+This block hash has to be compared on-chain, alongside the verification of the proof, to validate that the Steel proof is correct and to verify the integrity of RPC data.
+
+## What is a Steel Commitment?
+
+A commitment consists of two values: the block ID and the block digest.  The block ID encodes two values, the Steel version number and a block identifier in an encoding which is effectively: `abi.encodePacked(uint16 version, uint240 blockIdentifier)`.
+
+```solidity
+struct Commitment {
```

Comment:
Needs to be adapted according to https://github.com/risc0/risc0-ethereum/pull/281